network/src/metrics.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+use std::sync::Arc;
+
+pub trait NetworkMetrics {
+    fn network_concurrent_tasks(&self) -> &IntGaugeVec;
+}
+
+#[derive(Clone, Debug)]
+pub struct PrimaryNetworkMetrics {
+    /// The number of tasks running
+    pub network_concurrent_tasks: IntGaugeVec,
+}
+
+impl PrimaryNetworkMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        Self {
+            network_concurrent_tasks: register_int_gauge_vec_with_registry!(
+                "primary_network_concurrent_tasks",
+                "The number of concurrent tasks running in the network connector",
+                &["module", "network"],
+                registry
+            )
+            .unwrap(),
+        }
+    }
+}
+
+impl NetworkMetrics for PrimaryNetworkMetrics {
+    fn network_concurrent_tasks(&self) -> &IntGaugeVec {
+        &self.network_concurrent_tasks
+    }
+}
+
+impl Default for PrimaryNetworkMetrics {
+    fn default() -> Self {
+        Self::new(default_registry())
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct WorkerNetworkMetrics {
+    /// The number of tasks running
+    pub network_concurrent_tasks: IntGaugeVec,
+}
+
+impl WorkerNetworkMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        Self {
+            network_concurrent_tasks: register_int_gauge_vec_with_registry!(
+                "worker_network_concurrent_tasks",
+                "The number of concurrent tasks running in the network connector",
+                &["module", "network"],
+                registry
+            )
+            .unwrap(),
```

Comment:
see above re: the avoidable panic

Reply:
Similar to this https://github.com/MystenLabs/narwhal/pull/472#discussion_r928783850